src/components/editor/markdown-renderer/replace-components/todo-list/todo-list-replacer.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import { DomElement } from 'domhandler'
+import { ComponentReplacer, SubNodeConverter } from '../ComponentReplacer'
+
+const REGEX = /(\s*[-*] )(\[[ xX]])( .*)/
+
+export class TodoListReplacer implements ComponentReplacer {
+  content: string
+  onContentChange: (content: string) => void
+
+  constructor (content: string, onContentChange: (content: string) => void) {
+    this.content = content
+    this.onContentChange = onContentChange
+  }
+
+  handleCheckboxChange = (event: React.MouseEvent<HTMLInputElement, MouseEvent>): void => {
+    const checkbox = event.currentTarget
+    const checked = checkbox.checked
+    const startline = Number(checkbox.dataset.startline)
+    const lines = this.content.split('\n')
+    const line = lines[startline]
+    const results = REGEX.exec(line)
+    if (results) {
+      const before = results[1]
+      const after = results[3]
+      lines[startline] = `${before}[${checked ? 'x' : ' '}]${after}`
+      this.onContentChange(lines.join('\n'))
```

Comment:
Altering the full mark down code is a bit too heavy for an input component and for the markdown renderer. It should only render the code. I think it's a more elegant solution if the callback is used to say "the checkbox in line XYZ should be changed to TRUE/FALSE". Because then it's only an event and the parent component (or any other instance which maintainces the code) can decide what to do.